nightfall-deployer/contracts/Challenges.sol

```diff
@@ -85,34 +89,45 @@ contract Challenges is Stateful, Key_Registry, Config {
   (i.e. a trivial duplicate).
   */
     function challengeCommitment(
-        Block calldata block1,
-        Block calldata block2,
-        Transaction[] calldata transactions1,
-        Transaction[] calldata transactions2,
-        uint256 transaction1Index,
-        uint256 transaction2Index,
+        TransactionInfoBlock calldata transaction1,
+        TransactionInfoBlock calldata transaction2,
         uint256 commitment1Index,
         uint256 commitment2Index,
         bytes32 salt
     ) external onlyBootChallenger {
         checkCommit(msg.data);
+
+        // first, check we have real, in-train, contiguous blocks
+        state.areBlockAndTransactionReal(
+            transaction1.blockL2,
+            transaction1.transaction,
+            transaction1.transactionIndex,
+            transaction1.transactionSiblingPath
+        );
+        state.areBlockAndTransactionReal(
+            transaction2.blockL2,
+            transaction2.transaction,
+            transaction2.transactionIndex,
+            transaction2.transactionSiblingPath
+        );
+
+        require(
+            transaction1.blockL2.blockNumberL2 != transaction2.blockL2.blockNumberL2,
```

Comment:
I don't think this is correct - this would make duplicate commitments in different transactions but in the same block un-challengeable.

Reply:
The requirement if both transactions are in the same block is that they don't point to the same transaction (i.e. txIndex1 != txIndex2)